packages/taiko-client/proposer/proposer.go

```diff
@@ -69,6 +69,10 @@ type Proposer struct {
 	allowEmptyBlocks bool
 	initDone         bool
 	forceProposeOnce bool
+
+	// Bridge message monitoring
+	pendingBridgeMessages []*types.Transaction
+	bridgeMessageMutex    sync.Mutex
```

Comment:
Please prefer the use of `RWMutex` over a `Mutex`, so multiple go routines can be read from the messages slice at the same time.

nit: prefer smaller variable names in this case: `bridgeMsgMu`.